Season 02/homework/JeongMin/week 8/week8_14889.java

```diff
@@ -0,0 +1,74 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.StringTokenizer;
+
+public class week8_14889 {
+    public static int[][] power;
+    public static int[] visited;
+    public static int min = Integer.MAX_VALUE;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st;
+
+        // N값 받아서 배열 초기화
+        int N = Integer.parseInt(br.readLine());
+        power = new int[N][N];
+        visited = new int[N];
+
+        for (int i = 0; i < N; i++) {
+            st = new StringTokenizer(br.readLine());
+            for (int j = 0; j < N; j++) {
+                power[i][j] = Integer.parseInt(st.nextToken());
+            }
+        }
+
+        dfs(0, N,0);
+
+        System.out.println(min);
+    }
+
+    public static void dfs(int s, int N, int depth) {
+        //  수열의 길이가 N/2이 되었을 때
+        if (depth == N/2) {
+            calcDiff(N);
+            return;
+        }
+
+        for (int i = s; i < N; i++) {
+            if(visited[i] == 0) {
+                visited[i] = 1; // 노드 방문 표시
+                dfs(s+1, N,depth+1);
+                visited[i] = 0; // dfs가 끝나면 방문 표시 제거
+            }
+        }
```

Comment:
```suggestion
        for (int i = s; i < N; i++) {
            if(visited[i] == 0) {
                visited[i] = 1; // 노드 방문 표시
                dfs(i+1, N,depth+1);
                visited[i] = 0; // dfs가 끝나면 방문 표시 제거
            }
        }
```

ㅋㅋㅋ 찾았네요 눈빠지는줄~..
s+1 부터가아니고 i+1부터 진행해야 중복이 없겠죠?

Reply:
ㅎㅎ.. 시간 초과 때문에 눈물 박박 즙 짜고 있었는데 저거 때문이었다니... 
피드백 감사합니다...역시 최고의 멘토님..!